DuckDuckGo/TabViewController.swift

```diff
@@ -973,6 +984,14 @@ extension TabViewController: UIGestureRecognizerDelegate {
             reload(scripts: false)
         }
     }
+    
+}
 
+extension URL {
+    func toHttps() -> URL? {
+        let urlString = absoluteString
+        return URL(string: urlString.replacingOccurrences(of: "http", with: "https", options: .caseInsensitive, range: urlString.range(of: "http")))
+    }
```

Comment:
This can can create a malformed url if the scheme is already https.  I realise it's only going to be used when upgrading URLs, but that extension has global scope so could be dangerous. 

Using `URLComponents` would be safer:

```
        var components = URLComponents(url: self, resolvingAgainstBaseURL: false)
        components?.scheme = "https"
        return try? components?.asURL()
```

And since it's globally available, could have a unit test :)

Reply:
Oh good point, I just moved it, assumed it was solid without reading it. Will update and test.